test/e2e/upgrade_from_latest_test.go

```diff
@@ -24,6 +24,35 @@ func runUpgradeFromReleaseFlow(test *framework.ClusterE2ETest, latestRelease *re
 	test.DeleteCluster()
 }
 
+func runMulticlusterUpgradeFromReleaseFlowAPI(test *framework.MulticlusterE2ETest, latestRelease *releasev1.EksARelease, wantVersion anywherev1.KubernetesVersion, opts ...framework.ClusterE2ETestOpt) {
+	test.CreateManagementClusterForVersion(latestRelease.Version, framework.ExecuteWithEksaRelease(latestRelease))
+	test.RunConcurrentlyInWorkloadClusters(func(wc *framework.WorkloadCluster) {
+		wc.ApplyClusterManifest()
+		wc.WaitForKubeconfig()
+		wc.ValidateClusterState()
+	})
+	// Adding this manual wait because old versions of the cli don't wait long enough
+	// after creation, which makes the upgrade preflight validations fail
+	test.ManagementCluster.WaitForControlPlaneReady()
+	test.ManagementCluster.UpgradeCluster(opts)
+	test.ManagementCluster.ValidateCluster(wantVersion)
+	test.ManagementCluster.StopIfFailed()
+	cluster := test.ManagementCluster.GetEKSACluster()
+	// Upgrade bundle workload clusters now because they still have the old versions of the bundle.
+	test.RunConcurrentlyInWorkloadClusters(func(wc *framework.WorkloadCluster) {
+		wc.UpdateClusterConfig(
+			api.ClusterToConfigFiller(
+				api.WithBundlesRef(cluster.Spec.BundlesRef.Name, cluster.Spec.BundlesRef.Namespace, cluster.Spec.BundlesRef.APIVersion),
```

Comment:
So we are reading bundle for the management cluster because the management cluster has the latest but the workload cluster doesn't? Where do we define the one for the workload cluster again?

Reply:
Yes. It define the bundles for the workload cluster anywhere. After, the management and workload clusters are created, they have the same bundles. However, when the management cluster is updated, it then has the latest but the workload cluster still has the old bundles. The flow then continues to update the workload clusters with the bundles referenced by the management cluster.